ui/components/app/assets/token-list/token-list.tsx

```diff
@@ -76,17 +133,12 @@ export default function TokenList({
   ) : (
     <div>
       {sortedTokens.map((tokenData) => {
-        if (tokenData?.isNative) {
-          // we need cloneElement so that we can pass the unique key
-          return React.cloneElement(nativeToken as React.ReactElement, {
-            key: `${tokenData.symbol}-${tokenData.address}`,
-          });
-        }
```

Comment:
I think that if we include the `nativeTokens` as an array of react nodes in `TokenList`, rather than a fragment as we are now, we can retain this logic here.

The idea: when all tokens (including native tokens) get sorted and filtered together, each native token can be identified by the `isNative` prop. React can then render them as a `NativeToken` component, while still retaining it's sorted order within the list.